### PR TITLE
Validate against release branch

### PR DESCRIPTION
**Pull request by gal-berger**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/etc/issues/22674

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/validate/file_validator.py [hunk 1]
```diff
@@ -635,13 +630,22 @@ def validate_against_previous_version(self, no_error=False):
...
-                self._is_valid = prev_self_valid
+        compared_branch_name = self.prev_ver
+        if not self.prev_ver:
+            content_release_branch_id = self.get_content_release_identifier()
+            if not content_release_branch_id:
+                print_warning('could\'t get content\'s release branch ID. Skipping validation.')
+                return
+            else:
+                self.prev_ver = content_release_branch_id
+                compared_branch_name = 'release branch'
```
Comment (yaakovi) [inline demisto_sdk/commands/validate/file_validator.py:641]: a hardcoded string is not informative, use `content_release_branch_id`

File: demisto_sdk/commands/validate/file_validator.py [hunk 2]
```diff
@@ -663,3 +667,11 @@ def _is_py_script_or_integration(file_path):
             return True
 
         return False
+
+    def get_content_release_identifier(self):
+        try:
+            file_content = get_remote_file('.circleci/config.yml')
```
Comment (yaakovi) [inline demisto_sdk/commands/validate/file_validator.py:673]: use the `tag` option since this file might have changed in the current branch

Comment (gal-berger) [reply demisto_sdk/commands/validate/file_validator.py:673]: tag='master' ?

File: demisto_sdk/commands/validate/file_validator.py (hunk 1 above)
Comment (gal-berger) [reply demisto_sdk/commands/validate/file_validator.py:641]: done.

File: demisto_sdk/commands/validate/file_validator.py (hunk 2 above)
Comment (yaakovi) [reply demisto_sdk/commands/validate/file_validator.py:671]: Should be branch name

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 3]
```diff
@@ -245,3 +248,22 @@ def test_is_all_params_not_hidden(self, source, answer):
...
         validator = IntegrationValidator(structure)
         assert validator.is_all_params_not_hidden() is answer
+
+    with open(GIT_HAVE_MODIFIED_AND_NEW_FILES, "r") as test_params_file:
+        tests_params = json.load(test_params_file)
+    params = [
+        (None, tuple(set(i) for i in tests_params['data']['params_with_data']), '123456', True),
+        ('origin/master', tuple(set(i) for i in tests_params['data']['params_with_data']), '123456', True),
+        (None, tuple(set(i) for i in tests_params['data']['params_with_data']), '', True),
+        (None, tuple(set(i) for i in tests_params['data']['params_without_data']), '123456', True),
```
Comment (DeanArbel) [inline demisto_sdk/commands/validate/tests/validators_test.py:258]: please add a test for `file_validator._is_valid is False`

File: demisto_sdk/commands/validate/file_validator.py (hunk 2 above)
Comment (gal-berger) [reply demisto_sdk/commands/validate/file_validator.py:671]: done

File: demisto_sdk/commands/validate/tests/validators_test.py (hunk 3 above)
Comment (gal-berger) [reply demisto_sdk/commands/validate/tests/validators_test.py:258]: done

**Pull request merged**